ExomeDepth/make_BEDdetail.py

```diff
@@ -0,0 +1,365 @@
+    excluded_samples_file = open("{output}/{outputfile}_excluded_samples.txt".format(
+        output=args.outputfolder,
+        outputfile=args.outputfile
+        ),"w")
+
+    event_dic = {}
+    childs = 0
+    parents = 0 
+    for vcf in vcf_files:
+        exclude = False
+        sampleid = vcf.split("/")[-1].split("bam")[0].split("_")[2].rstrip(".")  
+        runid = "_".join(vcf.split("/")[-1].split("bam")[1].split("_")[1:-2])
+
+        if sampleid in merge_dic:  # Check if sample in specific run is merge sample. If so, exclude
+            for run in merge_dic[sampleid]:
+                if run == runid:
+                    excluded_samples_file.write("Sample {sampleid} run {runid} is excluded being merge sample\n".format(
+                        sampleid=sampleid,
+                        runid=runid
+                        ))
+                    exclude = True
+
+        if "giab" in sampleid.lower() or "control" in sampleid.lower(): # Remove GIAB and Control samples as these should not be included in the results
+            excluded_samples_file.write("Sample {sampleid} run {runid} is excluded being GIAB or Control sample\n".format(
+                sampleid=sampleid,
+                runid=runid
+                ))
+            exclude = True
+
+        if exclude == False:
+            with open(vcf, 'r') as vcf_lines:
+                """ Extract relevant fields from lines """
+                for line in vcf_lines:
+                    splitline = line.split()
+                    if "##" in line:
+                        if "EDreference" in line: 
+                            gender = line.split("=")[1].split("_")[1].rstrip()
+                            refset = line.split("=")[1].split("_")[2].rstrip()
+                    elif "#" in line:
+                        sampleid_vcf = str(splitline[9].rstrip())
+                        if sampleid != sampleid_vcf: # Check if sampleID in VCF is same as sampleID in VCF. If not: report and ignore
+                            excluded_samples_file.write("Sample {sampleid} run {runid} is excluded as sampleID of VCF file ({sampleid}) file is not the same as sampleID within VCF ({sampleid_vcf}\n)".format(
+                                sampleid=sampleid,
+                                runid=runid,
+                                sampleid_vcf=sampleid_vcf
+                                )) 
+                            break
+                    else:   
+                        infofields = splitline[7].split(";")
+                        formatfields = splitline[9].split(":")
+
+                        """ Sample stats """
+                        correl = formatfields[4]
+                        deldupratio = formatfields[8]
+                        totalcalls = formatfields[9]
+                        refsamples = formatfields[5] 
```

Comment:
Dit gaat er vanuit dat format veld vaste volgorde heeft, dat hoeft niet.

Reply:
Wel in de VCF files die ik maak voor de WES-CNV op basis van ExomeDepth

Reply:
Ga ik aanpassen met pyvcf